File: frontend/src/modules/seguridad/application/ListarRolesUseCase.ts
import { RolRepositoryPort } from '../domain/RolRepositoryPort';
import { Rol } from '../domain/Rol';

export class ListarRolesUseCase {
  constructor(private repo: RolRepositoryPort) {}
  async ejecutar(): Promise<Rol[]> {
    return this.repo.listar();
  }
}

File: backend/modulos/roles/aplicacion/ListarRolesUseCase.py
from typing import List
from ..dominio.RolRepositoryPort import RolRepositoryPort
from ..dominio.Entidades import Rol

class ListarRolesUseCase:
    def __init__(self, repo: RolRepositoryPort):
        self.repo = repo

    def ejecutar(self) -> List[Rol]:
        return self.repo.listar_roles()
